app/grandchallenge/cases/tasks.py

```diff
@@ -487,17 +493,31 @@ def _handle_image_relations(*, collected_images, upload_session):
         upload_session.archive.images.add(*collected_images)
 
 
-def _handle_unconsumed_files(
-    *, filepath_lookup, unconsumed_files, upload_session
+def _handle_raw_files(
+    *,
+    input_files: Set[Path],
+    consumed_files: Set[Path],
+    filepath_lookup: Dict[str, RawImageFile],
+    file_errors: Dict[Path, List[str]],
+    upload_session: RawImageUploadSession,
 ):
+    unconsumed_files = input_files - consumed_files
+
     errors = []
-    for unconsumed_filepath in unconsumed_files:
-        raw_file = filepath_lookup[str(unconsumed_filepath)]
-        error = raw_file.error or ""
+
+    for filepath in consumed_files:
+        raw_image = filepath_lookup[str(filepath)]
```

Comment:
When files are converted to tif, the newly created file is added to consumed files but of course do not exist in the filepath_lookup. I'm afraid I didn't add a test to check for that, sorry about that.

Reply:
Hmm - should they really be added to consumed files then? The images are still in the new_images and the associated files will be in new_image_files, right?

Reply:
Rather, are the two lines here (L423, L424) actually needed at all? https://github.com/comic/grand-challenge.org/pull/1415/commits/4af11423a99b02af501d4e1bae8ca02df4387d71

Reply:
yes, the associated files hold the source files that were used to create the tiff file. So file.mrxs and a bunch of .data files get converted to a single tif. It's the line above that adds the tiff to the consumed files. At that point, there's no check if the gc_file was a converted tif (in that case it doesn't need to be added) or original tif (then it should be added or it will be regarded as failed). I think this change should fix that: 
```
        if gc_file.associated_files:
            consumed_files |= set(f for f in gc_file.associated_files)
        else:
            consumed_files.add(gc_file.path)
```

Reply:
Great! That did the trick, we should add a test using the existing VMS and Mirax files in the resources directory.

Reply:
I ran 7809668 with individual zips of the complex formats and with convert_to_tiff.zip and everything worked well.